### PR TITLE
adds save and load single control

### LINKED ISSUE #13132: [GE] Save and load a single Control and its children as a JSON
Related forum discussion: https://forum.babylonjs.com/t/how-to-add-unknown-number-of-items-to-a-stackpanel-made-in-the-gui-editor/29564/8

> I don’t see a way to contribute to gui.babylonjs right now, but a feature I would love is the ability to output a single item or group as a .json file (think Snippits or Frames in deisgn programs like Sketch and Figma) and then just parse those smaller json snippits and not treat it as a fullscreen ui!

### PR DESCRIPTION
**Pull request by YifeiShi99**
This change resolves #13132 by adding the option to save and load single control in gui editor

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13812/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13812/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13812/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13812/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13812/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13812/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13812/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13812/merge#BCU1XR#0

File: packages/tools/guiEditor/src/components/commandBarComponent.tsx [hunk 1]
```diff
@@ -140,6 +140,16 @@ export class CommandBarComponent extends React.Component<ICommandBarComponentPro
...
                                     },
                                 },
+                                {
+                                    label: "Save selected contorl",
+                                    onClick: () => {
+                                        this.props.globalState.onSaveSelectedControl.notifyObservers();
+                                    },
+                                },
+                                {
+                                    label: "Load contorl",
```
Comment (carolhmj) [inline packages/tools/guiEditor/src/components/commandBarComponent.tsx:150]: nit: contorl -> control

**Review by carolhmj — CHANGES_REQUESTED** [review]
Thank you so much for the contribution! I have just a few word nits and a change needed on the loading part. 😄

File: packages/tools/guiEditor/src/components/commandDropdownComponent.tsx [hunk 2]
```diff
@@ -121,6 +121,15 @@ export class CommandDropdownComponent extends React.Component<ICommandDropdownCo
                                             )}
                                         </div>
                                     );
+                                } else if (m.fileButton && m.label === 'Load contorl') {
```
Comment (Popov72) [inline packages/tools/guiEditor/src/components/commandDropdownComponent.tsx:124]: nit: There' no need to test for `m.fileButton` as we are in the `else` of a `if (!m.fileButton)`.

Also, I feel a bit uneasy with testing a label in some logic code... Is there no way around it?

File: packages/tools/guiEditor/src/components/propertyTab/propertyTabComponent.tsx [hunk 3]
```diff
@@ -115,6 +120,22 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
         );
     }
 
+    /**
+     * Read loaded file
+     * @param file 
+     */
+    loadContorl(file: File) {
```
Comment (Popov72) [inline packages/tools/guiEditor/src/components/propertyTab/propertyTabComponent.tsx:127]: Another contorl => control.

You should perform a global search & replace to replace all occurrences of `contorl` with `control`.

File: packages/tools/guiEditor/src/diagram/workbench.tsx [hunk 4]
```diff
@@ -481,6 +481,25 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
...
+    /**
+     * Load Control from Json then select loaded Control
+     * @param serializationObject 
+     */
+    loadControlFromJson(serializationObject: any) {
+        const newSelection = [];
+        for (const control of serializationObject.controls) {
+            newSelection.push(Control.Parse(control, this.props.globalState.guiTexture));
+        }
+        const newGuiNode = this.props.globalState.workbench.appendBlock(newSelection[0]);
```
[the remark this answers is not in the document]
Comment (YifeiShi99) [reply packages/tools/guiEditor/src/diagram/workbench.tsx]: Hi @carolhmj, thanks for the suggestion, I see that this is causing only the first control is being loaded from the json with multiple controls saved. The appendBlock was moved to inside the loop to address this issue in the new commit. 

commit: ef8410cf2835f23dd355f8baf525218c9a4bca67

File: packages/tools/guiEditor/src/components/commandDropdownComponent.tsx [hunk 5]
```diff
@@ -121,6 +121,15 @@ export class CommandDropdownComponent extends React.Component<ICommandDropdownCo
                                             )}
                                         </div>
                                     );
+                                } else if (m.fileButton && m.label === 'Load contorl') {
+                                    return (
+                                        <FileButtonLineComponent
+                                            key={m.label}
+                                            label="Load contorl"
```
[the remark this answers is not in the document]
Comment (YifeiShi99) [reply packages/tools/guiEditor/src/components/commandDropdownComponent.tsx]: spelling fix is added in the new commit

commit: ef8410cf2835f23dd355f8baf525218c9a4bca67

File: packages/tools/guiEditor/src/components/commandBarComponent.tsx (hunk 1 above)
Comment (YifeiShi99) [reply packages/tools/guiEditor/src/components/commandBarComponent.tsx]: spelling fix is added in the new commit

commit: ef8410cf2835f23dd355f8baf525218c9a4bca67

File: packages/tools/guiEditor/src/components/commandBarComponent.tsx [hunk 6]
```diff
@@ -140,6 +140,16 @@ export class CommandBarComponent extends React.Component<ICommandBarComponentPro
                                         this.props.globalState.onSnippetLoadObservable.notifyObservers();
                                     },
                                 },
+                                {
+                                    label: "Save selected contorl",
```
[the remark this answers is not in the document]
Comment (YifeiShi99) [reply packages/tools/guiEditor/src/components/commandBarComponent.tsx]: spelling fix is added in the new commit

commit: ef8410cf2835f23dd355f8baf525218c9a4bca67

File: packages/tools/guiEditor/src/components/commandDropdownComponent.tsx (hunk 2 above)
Comment (YifeiShi99) [reply packages/tools/guiEditor/src/components/commandDropdownComponent.tsx]: Hi @Popov72, I think adding another property in "items" can also be use to check for button type for "Load control" button. New property in "items" added in commit to check button
commit: https://github.com/BabylonJS/Babylon.js/commit/ef8410cf2835f23dd355f8baf525218c9a4bca67

**Review by RaananW — APPROVED** [review]
Just run the formatter (`npm run format:fix`) and we are good to merge.

**Review by carolhmj — APPROVED** [review]
Great work! 🎉

**Comment by YifeiShi99** [comment]
@RaananW formatter (npm run format:fix) has been run in the latest commit 

commit: 51c29288db3a1970306d0f74f2378cd1c628a772

**Pull request merged**